### ISSUE TITLE
Error when executing "propel reverse"

### ISSUE THREAD

**Issue by itszacknotzack** [propelorm/Propel2#1562]
I am getting an error when calling "propel reverse" after updating to the lastest versions of composer and propel2. My system:

OS:  Windows 10 Pro
PHP: 7.4.6
Composer: 1.10.7
Propel: dev-master 897c198 (symfony: 5.1.0)

Here is the error I'm getting:

```
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function Symfony\Component\Config\Definition\Builder\TreeBuilder::__construct(), 0 passed in PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\PropelConfiguration.php on line 32 and at least 1 expected in PROJECTDIR\vendor\symfony\config\Definition\Builder\TreeBuilder.php:26
Stack trace:
#0 PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\PropelConfiguration.php(32): Symfony\Component\Config\Definition\Builder\TreeBuilder->__construct()
#1 PROJECTDIR\vendor\symfony\config\Definition\Processor.php(50): Propel\Common\Config\PropelConfiguration->getConfigTreeBuilder()
#2 PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\ConfigurationManager.php(197): Symfony\Component\Config\Definition\Processor->processConfiguration(Object(Propel\Common\Config\PropelConfiguration), Array)
#3 PROJECTDIR in PROJECTDIR\vendor\symfony\config\Definition\Builder\TreeBuilder.php on line 26

Fatal error: Uncaught ArgumentCountError: Too few arguments to function Symfony\Component\Config\Definition\Builder\TreeBuilder::__construct(), 0 passed in PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\PropelConfiguration.php on line 32 and at least 1 expected in PROJECTDIR\vendor\symfony\config\Definition\Builder\TreeBuilder.php:26
Stack trace:
#0 PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\PropelConfiguration.php(32): Symfony\Component\Config\Definition\Builder\TreeBuilder->__construct()
#1 PROJECTDIR\vendor\symfony\config\Definition\Processor.php(50): Propel\Common\Config\PropelConfiguration->getConfigTreeBuilder()
#2 PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\ConfigurationManager.php(197): Symfony\Component\Config\Definition\Processor->processConfiguration(Object(Propel\Common\Config\PropelConfiguration), Array)
#3 PROJECTDIR in PROJECTDIR\vendor\symfony\config\Definition\Builder\TreeBuilder.php on line 26
PHP Fatal error:  Uncaught ArgumentCountError: Too few arguments to function Symfony\Component\Config\Definition\Builder\TreeBuilder::__construct(), 0 passed in PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\PropelConfiguration.php on line 32 and at least 1 expected in PROJECTDIR\vendor\symfony\config\Definition\Builder\TreeBuilder.php:26
Stack trace:
#0 PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\PropelConfiguration.php(32): Symfony\Component\Config\Definition\Builder\TreeBuilder->__construct()
#1 PROJECTDIR\vendor\symfony\config\Definition\Processor.php(50): Propel\Common\Config\PropelConfiguration->getConfigTreeBuilder()
#2 PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\ConfigurationManager.php(197): Symfony\Component\Config\Definition\Processor->processConfiguration(Object(Propel\Common\Config\PropelConfiguration), Array)
#3 PROJECTDIR in PROJECTDIR\vendor\symfony\config\Definition\Builder\TreeBuilder.php on line 26

Fatal error: Uncaught ArgumentCountError: Too few arguments to function Symfony\Component\Config\Definition\Builder\TreeBuilder::__construct(), 0 passed in PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\PropelConfiguration.php on line 32 and at least 1 expected in PROJECTDIR\vendor\symfony\config\Definition\Builder\TreeBuilder.php:26
Stack trace:
#0 PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\PropelConfiguration.php(32): Symfony\Component\Config\Definition\Builder\TreeBuilder->__construct()
#1 PROJECTDIR\vendor\symfony\config\Definition\Processor.php(50): Propel\Common\Config\PropelConfiguration->getConfigTreeBuilder()
#2 PROJECTDIR\vendor\propel\propel\src\Propel\Common\Config\ConfigurationManager.php(197): Symfony\Component\Config\Definition\Processor->processConfiguration(Object(Propel\Common\Config\PropelConfiguration), Array)
#3 PROJECTDIR in PROJECTDIR\vendor\symfony\config\Definition\Builder\TreeBuilder.php on line 26
```

**Comment by Incognito** [comment]
There's some incompatibilities with Symfony 5 after they deprecated the TreeBuilder in https://symfony.com/blog/new-in-symfony-4-2-important-deprecations#deprecated-tree-builders-without-root-nodes .

Updating this dependency to work with sf5 is on our list of investigation topics. In the meanwhile, you may wish to explore if symfony 4.x is suitable for your project, or you may wish to override the classes (but that involves a fair bit more work).

**Comment by tsnork** [comment]
The thing is:

If you follow the documentation on propelorm.org and install via composer the required versions of propel's composer.json packages do not work with propel reverse. That's a bit of a bugger.

**Comment by PhilinTv** [comment]
@tsnork that's right. The chunk of SF5 related updates was too bit to tackle all as once, so bugs came in. https://github.com/propelorm/Propel2/pull/1563/files suppose to limit to only compatible SF5 components.

**Comment by dereuromark** [comment]
This should be fixed with https://github.com/propelorm/Propel2/pull/1582
which went into https://github.com/propelorm/Propel2/pull/1590

**Comment by zwhedbee** [comment]
`This should be fixed with #1582
which went into #1590`

Can confirm, thanks!

**Comment by Incognito** [comment]
We've merged a fix for the TreeBuilder incompatibility.
